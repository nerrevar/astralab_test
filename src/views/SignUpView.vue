<template>
  <div class="wrapper">
    <div class="title">
      Sign Up
    </div>
    <FormTemplate
      :formFields="formFields"
      submitText="Sign Up"
      @submit="submit($event)"
    />
    <div class="sign-in">
      Already have an account?
      <br />
      <router-link to="/sign_in">
        Sign In
      </router-link>
    </div>
    <div
      class="signup-error"
      v-if="errorText !== ''"
    >
      <span class="signup-error__text">
        {{ errorText }}
      </span>
      <img
        class="signup-error__close"
        src="@/assets/close.svg"
        alt="X"
        @click="errorText = ''"
      />
    </div>
  </div>
</template>

<script>
import { mapMutations, mapGetters } from 'vuex'

import FormTemplate from '@/components/FormTemplate.vue'

import {
  nameFormFieldParams,
  emailFormFieldParams,
  passwordFormFieldParams,
  repeatPasswordFormFieldParams
} from '@/utils'

export default {
  name: 'SignUpView',
  components: { FormTemplate },
  data: () => ({
    formFields: [
      nameFormFieldParams,
      emailFormFieldParams,
      passwordFormFieldParams,
      repeatPasswordFormFieldParams
    ],
    errorFields: [],
    errorText: '',
  }),
  computed: mapGetters(['getFirebaseApp']),
  methods: {
    ...mapMutations(['setUser']),
    submit (data) {
      this.getFirebaseApp.auth().createUserWithEmailAndPassword(data.email, data.password)
        .then(userCredential => {
          this.getFirebaseApp.auth().currentUser.updateProfile({ displayName: data.name })
          const { email } = userCredential.user
          this.setUser({
            user: {
              displayName: data.name,
              email,
            },
          })
          this.$router.push('/main')
        })
        .catch(error => {
          this.errorText = `Sign up error: ${ error.message }`
          setTimeout(
            () => {
              this.errorText = ''
            },
            10000
          )
        })
    },
  },
}
</script>

<style lang="sass" scoped>
.sign-in
  text-align: center
  margin-top: 2em

  & > a
    color: black

.signup-error
  +ellipse
  position: absolute
  right: 12px
  bottom: 12px
  left: 12px
  display: flex
  flex-flow: row nowrap
  justify-content: space-between
  background-color: #ff6683
</style>
