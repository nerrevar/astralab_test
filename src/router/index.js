import Vue from 'vue'
import VueRouter from 'vue-router'

import EnterView from '@/views/EnterView.vue'
import SignUpView from '@/views/SignUpView.vue'
import SignInView from '@/views/SignInView.vue'

Vue.use(VueRouter)

const routes = [
  {
    path: '/',
    name: 'EnterView',
    component: EnterView,
  },
  {
    path: '/sign_up',
    name: 'SignUpView',
    component: SignUpView,
  },
  {
    path: '/sign_in',
    name: 'SignInView',
    component: SignInView,
  },
  {
    path: '*',
    redirect: '/',
  }
]

const router = new VueRouter({
  mode: 'history',
  base: process.env.BASE_URL,
  routes,
})

export default router
