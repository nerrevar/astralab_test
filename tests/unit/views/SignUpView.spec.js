import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'

import router from '@/router'

import SignUpView from '@/views/SignUpView.vue'

const localVue = createLocalVue()

localVue.use(VueRouter)

describe('SignUpView.vue', () => {
  const vm = shallowMount(
    SignUpView,
    {
      localVue,
      router,
    }
  )

  it('renders', () => {
    expect(vm.element).toMatchSnapshot()
  })

  it('method submit', () => {
    expect(vm.vm.submit()).toBeUndefined()
  })
})
