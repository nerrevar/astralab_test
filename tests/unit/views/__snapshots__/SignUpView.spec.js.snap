// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpView.vue renders 1`] = `
<div
  class="wrapper"
>
  <div
    class="title"
  >
    
    Sign Up
  
  </div>
   
  <formtemplate-stub
    formfields="[object Object],[object Object],[object Object],[object Object]"
    submittext="Sign Up"
  />
   
  <div
    class="sign-in"
  >
    
    Already have an account?
    
    <br />
     
    <router-link-stub
      ariacurrentvalue="page"
      event="click"
      tag="a"
      to="/sign_in"
    >
      
      Sign In
    
    </router-link-stub>
  </div>
</div>
`;
